src/lib/y2storage/encryption_processes/pervasive.rb

```diff
@@ -63,19 +60,30 @@ def pre_commit(device)
         # API to allow sharing a new key among several volumes.
         @secure_key ||= generate_secure_key(device)
 
-        @zkey_cryptsetup_output = execute_zkey_cryptsetup(device)
-        return if @zkey_cryptsetup_output.empty?
+        master_key_file = "/etc/zkey/repository/#{@secure_key.name}.skey"
```

Comment:
The `SecureKey` class already contains the knowledge about the location of the repository (`/etc/zkey/repository` by default but maybe orverwritten by `ENV["ZKEY_REPOSITORY"]`) and its format. It would make sense to not duplicate that knowledge/logic here.